### PR TITLE
Edit set, delete set, better set page, date handling improvements

### PR DESCRIPTION
**Pull request by ParkerTonra**
# Summary of Changes:
Backend:
    Added update_collection function for updating beat collections (db.rs).
    Renamed update_beat to edit_beat for clarity (db.rs, main.rs).
    Introduced CollectionChangeset struct for managing collection updates (models.rs).

Frontend:
    Implemented handleEditSet and handleDeleteSet for managing collections (App.tsx).
    Enhanced BeatCollectionComponent to display collection details (venue, date, city, state, how many beats) (BeatCollection.tsx).
    Updated Sidebar and BeatTable to support editing collections (App.tsx, BeatTable.tsx).

Miscellaneous:

    Updated imports and function calls for renamed functions.
    Added CollectionChangeset TypeScript type definition (bindings.ts).
    Improved UI styles for collection info and conditional rendering (TableHeader isInCollection prop).

Finishing touches:
[x] navigate home after deleting a set
[x] make sidebar scroll better (fits all items)
[x] center the title vertically on set page
[x] fix icon for delete set button
[x] get confirmation before deleting a set

### CODE REVIEW

File: src/components/CollectionCard.tsx [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+        const submissionData: Partial<BeatCollection> = {
+            id: isCreating ? undefined : editedSet.id,
+            set_name: editedSet.set_name || undefined,
+            venue: editedSet.venue || undefined,
+            city: editedSet.city || undefined,
+            state_name: editedSet.state_name || undefined,
+            date_played: formattedDate || undefined,
+        };
+        console.log("formatted date", formattedDate);
+        console.log("submission data", submissionData);
```
Comment (daskimmel) [inline src/components/CollectionCard.tsx]: need these?

File: src/components/TableHeader.tsx [hunk 2]
```diff
@@ -140,13 +167,13 @@ export const TableHeader = ({
       )}
       {/* Reset to default settings button.
       TODO: This should be obfuscated final build */}
-      <button onClick={handleForceFirstTimeSetup}
+      {/* <button onClick={handleForceFirstTimeSetup}
```
Comment (daskimmel) [inline src/components/TableHeader.tsx]: do we plan on using this stuff later? just wondering if theres a good reason to keep it around

File: src/tests/Sidebar.test.tsx [hunk 3]
```diff
@@ -5,56 +5,79 @@ import { mockIPC } from '@tauri-apps/api/mocks';
...
-  },{
-    id: 2,
-    set_name: "Test Set 2",
-  }
+  id: 1,
+  set_name: "Test Set 1",
+}, {
+  id: 2,
+  set_name: "Test Set 2",
+}
```
Comment (daskimmel) [inline src/tests/Sidebar.test.tsx]: idk about this reformat lol but oh well

File: src/components/BeatCollection.tsx [hunk 4]
```diff
@@ -103,22 +121,50 @@ const BeatCollectionComponent: React.FC<BeatCollProps> = ({
...
+
+            <div className="whitespace-nowrap">
+              <span className="text-gray-400 mr-2">City:</span>
+              <span>{currentCollection.city || 'N/A'}</span>
+            </div>
+
+            <div className="whitespace-nowrap">
+              <span className="text-gray-400 mr-2">State:</span>
+              <span>{currentCollection.state_name || 'N/A'}</span>
+            </div>
```
Comment (daskimmel) [inline src/components/BeatCollection.tsx]: personally I think the text is a little large on the screen for these set attributes. I do really like how they hide on the small screen though.

Might be worthwhile to add total beats to the All Beats screen?

**Review by daskimmel — APPROVED** [review]
Just some small nits but overall works as expected 👍 good stuff

File: src/components/TableHeader.tsx (hunk 2 above)
Comment (ParkerTonra) [reply src/components/TableHeader.tsx]: no i just have commitment issues.

File: src/tests/Sidebar.test.tsx [hunk 5]
```diff
@@ -13,31 +13,49 @@ const mockCollections = [{
...
+    expect(screen.getByRole("button", { name: /Add New Set/i, hidden: true })).toBeInTheDocument();
     expect(screen.queryByPlaceholderText(/enter a name for a new set/i, {exact: true})).toBeInTheDocument();
   });
 
-  it("Allows for a new set to be added", async () => {
-    render(<Router><Sidebar collections={mockCollections} /></Router>);
+  it("Selecting a new set clears selected beats", () => {
+    const mockSetSelectedBeats = jest.fn();
+    render(<Router><Sidebar {...defaultProps}/></Router>);
+    screen.debug()
```
Comment (daskimmel) [inline src/tests/Sidebar.test.tsx:51]: can remove this

Comment (daskimmel) [reply src/tests/Sidebar.test.tsx:51]: although it may have come from my branch 😂

**Pull request merged**